pool/pool_machine.c

```diff
@@ -394,6 +442,19 @@ M0_INTERNAL void m0_poolmach_fini(struct m0_poolmach *pm)
 		if (pool_failed_devs_tlink_is_in(pd))
 			pool_failed_devs_tlink_del_fini(pd);
 	}
+
+	for (i = 0; i < state->pst_nr_nodes; ++i) {
+		cl = poolnode_clink(&state->pst_nodes_array[i]);
+		if (cl->cl_chan != NULL) {
+			obj = container_of(cl->cl_chan, struct m0_conf_obj,
+					   co_ha_chan);
+			M0_ASSERT(m0_conf_obj_invariant(obj));
+			m0_poolnode_clink_del(cl);
+			M0_SET0(cl);
+			m0_confc_close(obj);
```

Comment:
Looks to be repetitive functionality, add an helper function for it and use it.

Reply:
Done